In [1]:
# Outside imports
import os
import importlib
import warnings
warnings.filterwarnings('ignore')

import numpy as np
import pandas as pd
from sklearn.model_selection import train_test_split

from keras.preprocessing.text import Tokenizer
from keras.preprocessing.sequence import pad_sequences

Using TensorFlow backend.


In [22]:
import model
import train
import evaluate
import train_util
import data_util.data
import data_util.batcher
import data_util.config
import data_util.preprocess

importlib.reload(data_util.preprocess)
importlib.reload(train)
importlib.reload(model)
importlib.reload(evaluate)
importlib.reload(train_util)
importlib.reload(data_util.config)
importlib.reload(data_util.data)
importlib.reload(data_util.batcher)

from train import *
from evaluate import *
from model import *
from train_util import *
from data_util.data import *
from data_util.batcher import *
from data_util.preprocess import *

In [5]:
config.sum_path

'/Users/rowancassius/Desktop/capstone/LSTM_Summarizer/data/sum'

In [102]:
task_train = pd.read_csv(os.path.join(config.sum_path, 'task_train.tsv'), sep = '\t')
wiki_train = pd.read_csv(os.path.join(config.sum_path, 'wiki_train.tsv'), sep = '\t')

In [103]:
wiki_train.shape

(917986, 2)

In [66]:
from nltk.tokenize import sent_tokenize

In [67]:
sents = ['the', 'ground', 'is', 'wet']

In [68]:
['the', 'car'][:-1]

['the']

In [85]:
def split_wiki_text(text):
    sents = sent_tokenize(text)
    task_ = sents[-1]
    context_ = r'\s'.join(sents[:-1])
    return [task_, context_]

In [86]:
wiki_train[:1]

headline  \
0  \nBe careful in accepting a partial rent payment.   

                                                text  
0   If you accept partial payment, you can only p...

In [87]:
split_wiki_text(wiki_train.text[0])

 If you accept partial payment, you can only proceed with an eviction if the tenant agrees, in writing, to the terms and conditions of any partial payment agreement you reach with him or her. 


[' If you accept partial payment, you can only proceed with an eviction if the tenant agrees, in writing, to the terms and conditions of any partial payment agreement you reach with him or her.',
 '']

In [ ]:
wiki_train = wiki_train['headline'].join(wiki_train['text'].map(split_wiki_text).tolist())

In [94]:
# transform wiki data:
wiki_task_cont = wiki_train[:10000]['text'].map(split_wiki_text)

 If you accept partial payment, you can only proceed with an eviction if the tenant agrees, in writing, to the terms and conditions of any partial payment agreement you reach with him or her. 
 Lungworm infection is a serious lung infection caused by angiostrongylus vasorum. These worms invade the lungs and impair the body's ability to clot blood.Labored breathing, along with pale gums (because of hemorrhage), in a dog not routinely treated against lungworm, should raise suspicion of this life-threatening condition. A blood test is available that gives a "yes" "no" answer about lungworm infection.


 You want to make sure that it will look right. At the top of your Backstory, there is an eye. Click on that to check your progress. If you find that anything needs to be edited or cleared up a bit, just go back and fix it.


 Most grocery stores run weekly specials, which you can learn about through their weekly ads. When something is on sale, that's the time to stock up on it. Most protei


 After cooking your black beans, combine them with equal parts corn (canned or cut right off the cob) and freshly diced tomatoes, red onion and cilantro. Pour in a splash of red wine vinegar and savor with tortilla chips, pita bread or on top of a mound of melty nachos.Preparing corn and black bean salsa only requires a handful of ingredients and takes practically no time at all, making it perfect for easy dinners and last-minute potluck events.
Finish your salsa with other zesty additives like cubed mango or fresh jalapeno peppers.;

 Against the slight curve of a planting bed, the edging tool works fantastically. Simply spray paint your curved edge or lay down a garden hose onto the edge you want to create, and edge away. Of course, if you decide to use a garden hose as your guide, be extremely careful not to accidentally cut into it with your edging tool.


 Though it’s important to be open and honest during your assessment so you can get the right diagnosis, you may not be ready t

 Snap the motor clamps back into place. Tighten down the 1/4" snap clamp screws.
 Even a partial molar pregnancy cannot develop into a viable fetus and must be removed. The doctor will:Use a technique called dilation and curettage. This is done as an outpatient procedure so you will be able to go home the same day.
The doctor will dilate your cervix and insert a small vacuum through your vagina and into your uterus. The vacuum will remove the cystic tissue.


 Cover the pie with a piece of plastic wrap. Transfer the pie to the refrigerator and let it set for at least six hours. You can make this pie a full day in advance. Leave the pie chilling in the fridge for up to 24 hours before serving.When you're ready to serve the pie, garnish it with your favorite toppings and cut the pie into six or eight even slices.


 Set a budget for the overall trip so that you know what you can see and where you can go. Keep in mind that you'll have to pay for gas, lodging, food, and any entry fees that

 Once your Rocky Beach area is properly configured, you can encounter a Gible by walking around. It may take several battles for it to appear. Throw mud and then a Safari Ball to try to capture it.
 The world does not matter. Spin the Daily Spin if you have not already logged in that day. Check your jam-a-grams and Jammer Wall(if you are member). You are now ready to move to the next step.;

 These are plenty delicious as their own dish. Sprinkle with Parmesan or Asiago right away, so it melts over the tomatoes. Top with sprigs of fresh basil, Italian parsley, or pine nuts toasted for five minutes in a dry pan.


 Those who think positively double their chance to realize happiness.


 Slide the compressor off the stud. Remove the two valve springs and retainers.
 This is an important part of the process! Do not skip it! Licking your fingers adds a layer of water that protects your skin from getting burned


 Click on the black arrow next to your name in the top right.;

 If you haven't


 Enter "Poetry publishing" in the search field, and be prepared to sift through nearly 70 million results! Some are websites that publish poetry, some are poetry organizations, and some are just trying to make a buck. Always research a company before you submit your work to them, to be safe.


Google will generally sort results that are in your region first.


 Get some white, pumps with high heels, at least 4 inches (10.16 centimeters). Decorate them with candy dots (keep them in two rows of pink, 2 rows of blue, then 2 rows of yellow). You can also use puffy paint, regular paint, or buttons.


Don't like wearing high heels? Wear a different pair of choose, but keep the same theme: white with pink, blue, and yellow dots.


 Squares and rectangles are shapes frequently found in the modern style living room, as is any shape consisting of straight lines and angular edges. Perfect circles are also common, but rectangles with rounded edges or undefined shapes are not often found.


 Comme

 Only add as much to the juicer or blender as you can fit without having to cram it in. If making a juice with multiple fruits and/or vegetables, be sure to include equal parts of each produce into the juice as you work, and combine all the juice into one container to make sure that equal proportions of each produce type are spread throughout the entire large batch.
 If you want to toast the almonds (check that they're not already toasted), do so before proceeding.;

 Stay on Highway 299 through Blue Lake until you get to Willow Creek.
 Yawn widely to stretch your lips. Purse your lips and say "buh-buh-buh" while singing a scale. Try to make a good clear noise, not releasing too much air. If your lips keep losing the trill, stop and yawn, or put a finger in each cheek on either side of your mouth.Use this exercise regularly. The beginning of your routine is a good time for it, as it warms up your mouth.
Do your trills for a scale longer each time.


 It's the third tab from the left of


 Cook in the oven at 400ºF/200ºC until crisp.


Once cooked, crumble into smaller pieces.


 5 ÷ 4 = 1 R1. Place the quotient, 1, above the long division bar. Write the remainder above the first number of the dividend. Place a small 1 above the 5, to remind yourself that you had a remainder of 1 when you divided 5 by 4. 518 should now look like this: 5118









 Issues with your eye can cause serious, long time effects. Seek medical attention right away if you spot warning signs such as:


The inability to remove the contact from your eye
An embedded contact inside your eye
Abnormal vision
Pain, redness, or discomfort after the contact has been removed
 This is in the left sidebar of the file explorer window and will display all of the drives connected to your computer, whether or not they contain media.


The “This PC” option is the same as “My Computer” on older versions of Windows.


 One of the very first things you’ll want to do each morning is to spend 30 minutes planning out


 As you are reading through your essay, you might notice that you have gone off on a tangent, or that you’ve written in extraneous details, or that you’ve added fluff to your essay. Now is the time to remove any unnecessary information to help focus and improve the flow of your essay. Unnecessary extra information can confuse your reader, and you want to make sure that your essay is clear and concise.


 The valor form trains by hits, so for every time you land a hit on something in Valor Form you get one point of Valor Form Experience. To allow you to get an uninterrupted string of hits you should use weak enemies. The Shadows in the World That Never Was are a great and almost endless supply of free hits. Plus, they're almost everywhere in the city, so even if you do run out you'll be able to find some more.;

 Scrapes, rubs, and game trails are signs of deer frequenting a particular area. Be sure to hunt in shooting range from signs.


Scrapes are spots in the ground that bucks have


 Then, with your other hand, type 0191 on the numerical part of your keyboard. If this code doesn't work, try typing the numbers 168 while still holding down the Alt key.


 If your dog is not responding well to treats, try placing a favorite toy, or a brand new and especially tempting chew toy inside the crate.


 In the mammogram, your breast is placed on a platform and compressed with a paddle to even out the breast tissue, hold the tissue still during the x-ray, and allow for the use of a lower-energy x-ray.


You’ll feel pressure and may experience some discomfort during the mammogram, but this is just temporary.
A mammogram will be done on both breasts so the radiologist can compare both sides.


 Including regular exercise can help you manage and deal with anxiety. Studies have shown that exercise can have an immediate and long-lasting positive effect on anxiety.Aim for 150 mins of moderate-intensity aerobic activities each week or 75 minutes of vigorous intensity aerobic activ


 Then add warm ¼ cup olive oil to the pan again and place in minced garlic and red pepper flakes. This should sauté for a minute.


 When you were in conflict with your family over the fact that you didn’t visit them on vacation, you probably learned something that could help you avoid conflict or understand yourself better. For instance, perhaps this was the first time you ever did something your family disapproved of. This might make you realize that you are a strong person.Whatever your case may be, reflect on what you learned about yourself and your family through the conflict.


 Depending on your preference and intended use, you may leave your kohlrabi bulb whole or cut it into ½” cubes.
 (28g). Most lobster tails require between 5 and 12 minutes.


 When the bottom of the sandwiches are golden brown, use a spatula to carefully flip them over in the pan. Allow them to cook on the other side for another 3 to 5 minutes or until it is also golden brown and the cheese is melted.Serv


 Understand and respect if parents don't want her eating something in particular - for example, very sugary foods or processed foods. Again, don't forget her formula.


 When you plan to eat other foods without vitamin E in them that day, take a supplement. When you plan to eat foods rich in vitamin E, do not take any supplements containing this vitamin that day.


 Keep track of your ducks through proper identification. In a small group you may recognize individual ducks but if you cannot, it may be helpful to use wing or leg bands to mark individual birds.


 Lanterns are designed to hang in a certain way so as to avoid overheating. Always follow the manufacturer's specification.


 This is the most common way of saying "thank you" in Russian, and it's appropriate for most situations.Pronounce "cпасибо" as: spa-see-ba

While common, this phrase is actually considered the least formal way of saying “thank you” in Russian. You can probably get through most conversations using this phr


 As you enter your delivery stride, try to get your whole body to face reasonably side on.


 Hide ‘’nothing’’ about your life. If you had a disgusting surgery, tell it in all its gory details. Tell him explicit stories about your exes, family life, and friendship troubles.


 Infants, elderly persons and persons with chronic or debilitating illnesses such as diabetic renal failure are particularly vulnerable to complications of dehydration caused by cholera. If anyone in these groups shows signs of cholera, need urgent admission to the hospital for rehydration and close monitoring. Anyone with severe dehydration will also have much better odds of recovery if they get to a hospital immediately for IV fluid restoration.Seek immediate medical attention for children of all ages with severe dehydration.


 Having bipolar disorder creates limitations your friend must live with, and in order to effectively help them it’s important that you understand and respect them as well. A person suffe


 The Ethernet cable must be plugged into an open Ethernet port on your modem, and to the port on your Belkin router labeled “WAN / Internet.”


 Everyone makes mistakes. Part of becoming a mature adult is taking responsibility for the mistakes that you make. By doing this, you will begin to win over the trust and respect of other adults. Own up to your mistakes and then do what you can to correct it.For example, you should admit to backing into your neighbor’s mailbox, even if nobody witnessed it. Inform your neighbor of your mistake and repair or replace their mailbox for them. This will show that you are taking responsibility and holding yourself accountable for your actions.


 If you want to use another drive letter, put it in then press enter.


 If you live with other people or have children in your home, keep the medication in a secure, locked location.This can safeguard against misuse, abuse, or accidental doses.


If medication is given to a child, administer single doses eac


 The time may vary. However, you can tell when the cake is done if you poke a toothpick into the center, and it comes out clean.


 Most states have created exemption schemes for bankruptcy. You should read your state law to see what exemptions are available.


You can find your state law by searching the Internet. Type “your state” and “bankruptcy exemptions” into your favorite web browser. Then scroll through the results.
You might also want to stop into your nearest law library. It should be either at the courthouse or at your nearest law school. Tell the librarian that you are looking for your state’s bankruptcy exemptions. The librarian should help you find the information.


 Just like with a real horse, an imaginary can be hard work. Take it slowly and let your imagination soar!


 Choose a reliable printer to increase the quality of your card. You may find a printer by entering postcard into your favorite search engine. You may also want to read customer reviews.


 Your expen

 Though most people may not think of their diet as having a negative effect on the skin, some research suggests that an unhealthy diet, including a diet rich in processed foods, refined carbohydrates, and unhealthy fats can cause premature signs of aging and damage the skin.
 Give him brown hair and a full beard. Make him relatively muscular. Dress him in casual clothing. Use the traits Family-Oriented, Handy, Good, Artistic, and Friendly.


 Obey your parents if you are a kid, and do your chores and homework. If you are all grown up, call your parents once in a while.


 The two negative signs meeting means the result is positive.

Example:-12/-4=3.


 Select Murkrow from the list of Pokémon, and press the console’s A button to confirm using the Dusk Stone. Watch as Murkrow evolves to Honchkrow.


Do not cancel the evolution or you’ll waste a stone.


 After submitting your application through your PayPal account by clicking or tapping the button that says “Agree and Continue,” you wi


 However, don't spend the whole night dwelling and stressing. If you've prepared well in the previous weeks, you won't need to do much last-minute rehearsal. After a few practice runs, put your script or notes away and try to clear your head.


 Grab some thawed, raw pork chops, a sharp paring knife, preserves, and a spoon. Slice lengthwise down the side of the chops, opening a pocket inside. Scoop your favorite preserves into the inside of the meat. Bake the pork chops as you normally would, or for about 30 minutes at 365℉ (185℃).Internal temperature for pork is recommended at 145℉ (63℃).
Cherry, apple, fig, or any kind of pepper preserves go well with pork, but don't be afraid to experiment with other preserves.


 Dress in tights and a sweatshirt.


The tights and sweatshirt should match the color of the frosting. If you used white frosting, then wear white clothes. If you used chocolate frosting, wear brown clothes.
Alternatively, you could wear flesh-tone tights instead of colore

 Level four headings are used for substantial subsections within an already divided section. In other words, use a level four heading to title a subsection below a level three heading, within the level three heading's subsection.


Level four headings are written in italics, without any bold font. The text should be centered within the body of the paper.
 Try arriving for your interview about 10 minutes early. This will give you some breathing time to mentally prepare for the interview. If the company has a receptionist or assistant, you can let him or her know that you are there for an interview while you wait.


If your interview takes place over the phone or a videoconference, have everything set up in a quiet, private, professional location at least 10 minutes before the interview is supposed to start.


 Educational brochures should be 40-50% white space to make them easier to read. You probably have a lot of information you want to include, but brochures aren’t books. They should


 This will keep the 'talk' button held down for continuous broadcasting of music or sound to the chat room and anyone who has the voice or sound feature enabled. The 'hands free' feature does not work very well. Some programs that work with Yahoo are equipped with a mic locker. You can download these on the internet, but always run a virus checker on them, since these programs are not supported by Yahoo.


 Hold your cat against you, facing away from you (wrapped in a towel to restrict its movement, if necessary). With one hand, steady your cat’s head and tilt it upwards. Make sure that your cat stays leaned up against you. 
 Shoot the wheels, shoot the driver or shoot at the gas tank and set the car on fire. Doing so will cause the thieves to exit the van. Once they do, kill them.


Alternatively, you could toss a sticky bomb onto the van and detonate it.


 If you tend to get mad at your mom due to certain words or circumstances, your anger is most likely situational; try avoiding t





Ring bound, spiral bound, perfect bound or saddle stitched books are accepted.
Make sure all pages are cut. Otherwise the person at Google Books scanning the book may skip pages.
Be certain no pages are stuck together or pages may be skipped.


 Yes, it may sound like a lot of hard work but even if you sell the feather for just 10gp each, that's 1500gp already. If you have 200 feathers, that's 2000gp, or 2k. It's a lot of money when you think that all you've done is killed a few chickens. And you'll be leveling up at the same time!


 Use a small amount of the mixture on the cloth and be extremely gentle. You want to use the same basic principle as the quick-fix method, gently dabbing and letting the cleaner do the work. Don't make it worse by scrubbing.









 Chickweed is said to begin healing on contact with your skin. This soothing little herb can be added to an herbal salve that is helpful in healing chafing, scrapes and other minor skin irritations.Leave the salve on the a

In [97]:
wiki_task_cont.tolist()

[[' If you accept partial payment, you can only proceed with an eviction if the tenant agrees, in writing, to the terms and conditions of any partial payment agreement you reach with him or her.',
  ''],
 ['A blood test is available that gives a "yes" "no" answer about lungworm infection.',
  " Lungworm infection is a serious lung infection caused by angiostrongylus vasorum.\\sThese worms invade the lungs and impair the body's ability to clot blood.Labored breathing, along with pale gums (because of hemorrhage), in a dog not routinely treated against lungworm, should raise suspicion of this life-threatening condition."],
 ['If you find that anything needs to be edited or cleared up a bit, just go back and fix it.',
  ' You want to make sure that it will look right.\\sAt the top of your Backstory, there is an eye.\\sClick on that to check your progress.'],
 ['Most protein can be frozen and saved for later, for instance, so if your store is running a sale on boneless, skinless chicken br

In [106]:
wiki_train[:1000].join(wiki_task_cont.tolist(), columns=['TaskSentence', 'Context'])

TypeError: join() got an unexpected keyword argument 'columns'

In [90]:
wiki_train.shape

(10000,)

In [9]:
# Process the the data
def prep_data(df):
    df['Context'] = df['Context'].map(lambda x: article_process_text(x))
    df['TaskSentence'] = df['TaskSentence'].map(lambda x: article_process_text(x))
    df['Summary'] = df['Summary'].map(lambda x: summary_process_text(x))
    return df
train_data = prep_data(train_data)
# test_data = prep_data(test_data)

In [91]:
vocab = Vocab(os.path.join(config.vocab_path, 'vocab3.txt'))

Duplicated word in vocabulary file: "
Duplicated word in vocabulary file: "
Duplicated word in vocabulary file: "
Duplicated word in vocabulary file: "
Duplicated word in vocabulary file: "
Duplicated word in vocabulary file: "
Duplicated word in vocabulary file: "
Duplicated word in vocabulary file: "
Duplicated word in vocabulary file: "
Duplicated word in vocabulary file: "
Duplicated word in vocabulary file: "
Duplicated word in vocabulary file: "
Duplicated word in vocabulary file: "
Duplicated word in vocabulary file: "
Duplicated word in vocabulary file: "
Duplicated word in vocabulary file: "
Duplicated word in vocabulary file: "
Duplicated word in vocabulary file: "
Duplicated word in vocabulary file: "
Duplicated word in vocabulary file: "
Duplicated word in vocabulary file: "
Duplicated word in vocabulary file: "
Duplicated word in vocabulary file: "
Duplicated word in vocabulary file: "
Duplicated word in vocabulary file: "
Duplicated word in vocabulary file: "
Duplicated w

In [29]:
opt = Namespace(
    train_mle = "yes", 
    train_rl = "no", 
    mle_weight = 1., 
    load_model = None,
    new_lr = None, 
    rl_weight = 0.)

wiki_task_batcher = TaskBatcher(
    examples=wiki_train.to_dict('records'),
    vocab=vocab,
    mode='train',
    batch_size=32,
    single_pass=False
)

# val_task_batcher = TaskBatcher( # Batching obj
#     examples=test_data.to_dict('records')[:200],
#     vocab=vocab, 
#     mode='train', 
#     batch_size=50, 
#     single_pass=False
# )



In [ ]:
# train_processor = TaskTrain(vocab, task_batcher, opt, TaskModel, val_task_batcher)
train_processor = TaskTrain(vocab, task_batcher, opt, TaskModel)

# load pre-trained embedding weights
train_processor.model.load_embeddings("embedding_43861_200.tar")

In [30]:
# train
mle_losses = train_processor.trainIters(n_iters=20, report_every=1, save_every = 5)

iter: 1 mle_loss: 6.868 mle_loss_val: -100.0000
iter: 2 mle_loss: 6.675 mle_loss_val: -100.0000
iter: 3 mle_loss: 7.420 mle_loss_val: -100.0000
iter: 4 mle_loss: 6.644 mle_loss_val: -100.0000
iter: 5 mle_loss: 6.621 mle_loss_val: -100.0000
model saved at: 
 data/saved_models/0000005.tar
iter: 6 mle_loss: 5.994 mle_loss_val: -100.0000
iter: 7 mle_loss: 5.094 mle_loss_val: -100.0000
iter: 8 mle_loss: 4.194 mle_loss_val: -100.0000
iter: 9 mle_loss: 4.651 mle_loss_val: -100.0000
iter: 10 mle_loss: 3.671 mle_loss_val: -100.0000
model saved at: 
 data/saved_models/0000010.tar
iter: 11 mle_loss: 3.678 mle_loss_val: -100.0000
iter: 12 mle_loss: 4.053 mle_loss_val: -100.0000
-------------------Keyboard Interrupt------------------


NameError: name 'exit' is not defined

In [24]:
mle_losses

{'train': [(1, 6.6681809425354),
  (2, 7.459186553955078),
  (3, 6.613460540771484),
  (4, 6.715197563171387),
  (5, 5.890581130981445),
  (6, 5.120395660400391)],
 'val': []}

In [32]:
opt_retrain = Namespace(
    train_mle = "yes", 
    train_rl = "no", 
    mle_weight = 1., 
    load_model = '0000010.tar',
    new_lr = None, 
    rl_weight = 0.)

train_processor = TaskTrain(vocab, task_batcher, opt_retrain, TaskModel)

mle_losses = train_processor.trainIters(n_iters=5, report_every=1, save_every = 5)

Loaded model at data/saved_models/0000010.tar
iter: 1 mle_loss: 3.565 mle_loss_val: -100.0000
-------------------Keyboard Interrupt------------------


NameError: name 'exit' is not defined

In [ ]:
mle_losses

Decoding Time

In [28]:
len(train_data.to_dict('records') + test_data.to_dict('records'))

2820

In [26]:
dat.shape

(2820, 11)

In [29]:
load_model = os.path.join(config.log_root, "data/saved_models_2/0000360.tar") # model directory

# new batcher for evaluation
task_batcher = TaskBatcher( # Batching obj
    examples=train_data.to_dict('records') + test_data.to_dict('records'),
    vocab=vocab, 
    mode='train', 
    batch_size=282,    # 188, 
    single_pass=True)

eval_processor = TaskEvaluate(vocab, task_batcher, TaskModel, load_model) # Evaluation object

In [30]:
2256/8

282.0

In [31]:
train_data.shape

(2256, 11)

In [32]:
def count_parameters(model):
    return sum(p.numel() for p in model.parameters() if p.requires_grad)

In [33]:
count_parameters(eval_processor.model)

14911184

In [34]:
decoded_sents, ref_sents, task_sents, context_sents = eval_processor.evaluate_batch()

Summarizing Batch...
Summarizing Batch...
example_generator completed reading all examples. No more data.


Exception in thread Thread-9:
Traceback (most recent call last):
  File "/Users/rowancassius/Desktop/capstone/LSTM_Summarizer/data_util/batcher.py", line 448, in text_generator
    example = next(example_generator)
StopIteration

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/Users/rowancassius/opt/anaconda3/lib/python3.7/threading.py", line 926, in _bootstrap_inner
    self.run()
  File "/Users/rowancassius/opt/anaconda3/lib/python3.7/threading.py", line 870, in run
    self._target(*self._args, **self._kwargs)
  File "/Users/rowancassius/Desktop/capstone/LSTM_Summarizer/data_util/batcher.py", line 429, in fill_example_queue
    context, task, summary = next(input_gen) # read the next example from file. article and abstract are both strings.
RuntimeError: generator raised StopIteration



Summarizing Batch...
Summarizing Batch...
Summarizing Batch...
Summarizing Batch...
Summarizing Batch...
Summarizing Batch...
Summarizing Batch...
Summarizing Batch...
INFO:tensorflow:Finished reading dataset in single_pass mode.


In [35]:
len(ref_sents)

2820

In [36]:
len(decoded_sents)

2820

In [37]:
scores = Rouge().get_scores(decoded_sents, ref_sents, avg = True)

In [38]:
scores

{'rouge-1': {'f': 0.6020702789916959,
  'p': 0.7667154319281988,
  'r': 0.5381503106904498},
 'rouge-2': {'f': 0.3904763855618124,
  'p': 0.5170605369807497,
  'r': 0.35230873911421756},
 'rouge-l': {'f': 0.5992272203650424,
  'p': 0.7633533920767971,
  'r': 0.5348177231306076}}

In [22]:
pd.set_option('display.max_colwidth', -1)

In [42]:
df = pd.DataFrame({'task': task_sents,'ref': ref_sents, 'decoded': decoded_sents})
df.sample(10)

task  \
2457  Please send a package of article reprints to the attention of Joan Payne at the following address:                                                                                         
2536  Please invite Steve Schwarzbach                                                                                                                                                            
2751  Check out the E-Flyer below and let me know what's up.                                                                                                                                     
2683  Please remove the Personal Expense icon from all personal expense entries and submit the expense report once again for payment.                                                            
1843  As soon as you get confirmation that your request has been completed, please let Edmund know so he can update/replace the London documents in the Financial Trading Agreements Database.   
1391  Please take me off of this distribution.                                                                                                                                                   
1047  Tom: Please forward your requisite email request to Donna Lowry, Rick Carson and Bill Bradford with your request and reasons.                                                              
2025  To register for the Energy Town Hall in Dallas on Wednesday, Nov.28, please simply reply to this email with your contact information.                                                      
536   Please take a look at this email ....                                                                                                                                                      
2533  please call after reading                                                                                                                                                                  

                                                                                   ref  \
2457  send package of article reprints to Joan Payne                                     
2536  invite Steve Schwarzbach                                                           
2751  look at the e - flyer                                                              
2683  remove personal expense icon from entries and resubmit expense report              
1843  confirm request completion with Edmund                                             
1391  remove SENDER from distribution                                                    
1047  forward requisite email request to Donna Lowry , Rick Carson , and Bill Bradford   
2025  reply to email with contact information to register for energy town hall           
536   examine this email                                                                 
2533  call SENDER                                                                        

                              decoded  
2457  send package of article          
2536  invite Steve Schwarzbach         
2751  check out e - flyer              
2683  remove personal expense          
1843  update SENDER                    
1391  take SENDER off of distribution  
1047  forward requisite request        
2025  reply to SENDER                  
536   look at email                    
2533  call SENDER

In [43]:
df.to_csv(os.path.join(config.log_root, 'data/test_results_2.csv'), sep = '\t', header=True, index=False)

In [44]:
'packet' in vocab._word_to_id

True

In [45]:
'interview' in vocab._word_to_id

True

Checking certain words for presence in the vocab:

In [46]:
words = ["ensure", "indicate", "turn", "open", "add"]

In [47]:
[w in vocab._word_to_id for w in words]

[True, True, True, True, True]

In [48]:
action_words = train_data['Summary'].map(lambda x: x.split(' ', 1)[0])

In [49]:
action_words.dtype

dtype('O')

In [50]:
in_vocab = action_words.map(lambda w: w in vocab._word_to_id)

In [51]:
len(in_vocab)

2256

In [52]:
in_vocab.sum()

2240

In [53]:
action_words[~in_vocab]

1354    revier  
76      relpy   
2190    briing  
1213    exame   
2192    email/  
126     reiview 
990     hande   
1366    foraard 
1988    confirm/
14      reissue 
25      exectute
2627    meesage 
1425    complute
416     pring   
1914    verity  
492     helo    
Name: Summary, dtype: object

In [54]:
test_data[~in_vocab]

IndexingError: Unalignable boolean Series provided as indexer (index of the boolean Series and of the indexed object do not match).

2240 of 2256 verbs to start summaries are in the vocab

In [ ]:
# train
action_words[~in_vocab]

In [ ]:
train_data[~in_vocab]['Labeler'].value_counts()

In [ ]:
train_data[~in_vocab]['Summary']

Look at Worst summaries

In [55]:
def score(r):
    scores = Rouge().get_scores(r.decoded, r.ref)[0]
    return [scores['rouge-1']['f'], scores['rouge-2']['f'], scores['rouge-l']['f']]

In [56]:
scored_examples = [score(r) for r in df.itertuples()]

In [57]:
df = pd.concat([df, pd.DataFrame(scored_examples, columns=['Rouge-1', 'Rouge-2', 'Rouge-L'])], axis=1)

In [58]:
df

task  \
0     If you would like to take advantage of this, please fax a completed registration form, along with a brief covering note referring to this email, and I will ensure that our administration depart handle the relevant paper work.   
1     If you are within 12 months of graduating and haven't already sent me an email, please reply letting me know you are eligible and interested in participating in registering for the fall 01 - spring 02 OCR program.?              
2     Please reallocate mtr 1517 with these changes; sale of 10,000 on deal 93804, purchase of 4,000 on 148851.                                                                                                                           
3     If you are not an intended recipient of this message, please do not read, copy or disclose the contents of this communication to others and please notify the sender at 713-2077-414 .                                              
4     Could you please take a look at the document it's a very quick read , call me if you need more information, and then vote for which option you would prefer?                                                                        
...                                                                                                                                                            ...                                                                        
2815  Can you call me?                                                                                                                                                                                                                    
2816  Please respond to EnergyWriter@aol.com                                                                                                                                                                                              
2817  Please respond to loughrid@rice.edu                                                                                                                                                                                                 
2818  Please respond to Whitepapers                                                                                                                                                                                                       
2819  please print out                                                                                                                                                                                                                    

                                            ref                      decoded  \
0     fax complete registration form to SENDER   fax completed form            
1     reply to SENDER about OCR program          reply to SENDER               
2     reallocate mtr 1517                        reallocate with changes       
3     do not read contents of this commuication  notify read at 713-2077-414   
4     read document and vote                     call SENDER                   
...                      ...                             ...                   
2815  call SENDER                                call SENDER                   
2816  respond to energywriter@aolcom             respond to [UNK]              
2817  respond to loughrid@riceedu                respond to [UNK]              
2818  respond to whitepapers                     respond to whitepapers        
2819  print                                      print out out                 

       Rouge-1   Rouge-2   Rouge-L  
0     0.444444  0.000000  0.444444  
1     0.666667  0.571429  0.666667  
2     0.333333  0.000000  0.333333  
3     0.181818  0.000000  0.181818  
4     0.000000  0.000000  0.000000  
...        ...       ...       ...  
2815  1.000000  1.000000  1.000000  
2816  0.666667  0.500000  0.666667  
2817  0.666667  0.500000  0.666667  
2818  1.000000  1.000000  1.000000  
2819  0.500000  0.000000  0.666667  



In [62]:
df_bad = df[df['Rouge-1'] < 0.1]

In [65]:
df_bad

task  \
4     Could you please take a look at the document it's a very quick read , call me if you need more information, and then vote for which option you would prefer?                                           
30    Please take a few minutes today to provide much needed and confidential comments regarding Enron and your work environment.                                                                            
76    Peter Feltman/ENRON@enronXgate 04/10/2001 02:31 PM To: Mark Greenberg/NA/Enron@ENRON cc: Mathis Conner/ENRON@enronXgate Subject: Global Crossing's comments to the Enron NDA - Please responed ASAP!   
78    Can you pass up this opportunity??                                                                                                                                                                     
85    Please look over it carefully and if there is language you are going to provide please do it post haste.                                                                                               
...                                                                                                        ...                                                                                               
2702  -bcc: X-Folder: Tanya Jones Dec2000 Notes Folders All documents X-Origin: JONES-T X-FileName: tjones.nsf                                                                                               
2704  After you read through it, please get back with me.                                                                                                                                                    
2766  I assume that this request relates to corporate functions under Rick.                                                                                                                                  
2778  Can you see if this is the correct CP.                                                                                                                                                                 
2810  -Origin: Taylor-M X-FileName: mtaylor.nsf                                                                                                                                                              

                                        ref                        decoded  \
4     read document and vote                 call SENDER                     
30    comment on work conditions             provide today to confidential   
76    respond to SENDER                      [UNK]                           
78    get back to SENDER                     pass up                         
85    review it and respond with language    look post                       
...                                   ...          ...                       
2702  send to SENDER after recieving         t notes                         
2704  respond to SENDER after reading email  read back                       
2766  submit by tomorrow                     corporate request               
2778  check this for correctness             review correct                  
2810  contact Fred Lagrasta                  Taylor-M X-FileName             

      Rouge-1  Rouge-2  Rouge-L  
4     0.0      0.0      0.0      
30    0.0      0.0      0.0      
76    0.0      0.0      0.0      
78    0.0      0.0      0.0      
85    0.0      0.0      0.0      
...   ...      ...      ...      
2702  0.0      0.0      0.0      
2704  0.0      0.0      0.0      
2766  0.0      0.0      0.0      
2778  0.0      0.0      0.0      
2810  0.0      0.0      0.0      

[143 rows x 6 columns]

In [64]:
import urllib.request

In [ ]:
urllib.request.urlretrieve()